crates/orchestrator/src/client.rs

```diff
@@ -156,6 +161,33 @@ pub struct MultiValidatorArgs {
     pub network_config_file: Option<String>,
 }
 
+/// Asynchronously retrieves a `NetworkConfig` from an orchestrator.
+/// The retrieved one includes correct `node_index` and peer's public config.
+///
+/// # Errors
+/// If we are unable to get the configuration from the orchestrator
+pub async fn get_complete_config<K: SignatureKey>(
```

Comment:
I needed to decouple `OrchestratorClient`. This method didn't really *need* to live on `NetworkConfig` anyway since it creates a network config, and has a side effect output with the source value.